src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -495,6 +496,7 @@ public AzureVMDeploymentInfo createDeployment(
             copyVariableIfNotBlank(tmp, properties, "imageVersion");
             copyVariableIfNotBlank(tmp, properties, "osType");
             putVariable(tmp, "ephemeralOSDisk", Boolean.toString(ephemeralOSDisk));
+            putVariable(tmp, "encryptionAtHost", Boolean.toString(encryptionAtHost));
```

Comment:
adding this variable won't do anything, you need to either modify all the arm templates to add something that uses this variable, or add some code that inserts it into the arm template around here somewhere.